actionview/lib/action_view/helpers/tags/collection_helpers.rb

```diff
@@ -79,6 +79,39 @@ def render_collection #:nodoc:
             yield item, value, text, default_html_options.merge(additional_html_options)
           end.join.html_safe
         end
+
+        def render_collection_for(builder_class, &block) #:nodoc:
+          options = @options.stringify_keys
+          rendered_collection = render_collection do |item, value, text, default_html_options|
+            builder = instantiate_builder(builder_class, item, value, text, default_html_options)
+
+            if block_given?
+              @template_object.capture(builder, &block)
+            else
+              render_component(builder)
+            end
+          end
+
+          # Append a hidden field to make sure something will be sent back to the
+          # server if all radio buttons are unchecked.
+          if options.fetch('include_hidden', true)
+            rendered_collection + hidden_field
+          else
+            rendered_collection
+          end
+        end
+
+        def hidden_field #:nodoc:
```

Comment:
Hello @maurogeorge – I'm :+1: with this PR :grinning: 

Since you added `def hidden_field`, could the one [already defined in collection_check_boxes.rb](https://github.com/rails/rails/pull/18303/files#diff-ad5a6748d86270f717368d13026b475bR27) be removed?